crates/ruff/src/codes.rs

```diff
@@ -807,6 +807,7 @@ pub fn code_to_rule(linter: Linter, code: &str) -> Option<(RuleGroup, Rule)> {
         (Ruff, "014") => (RuleGroup::Nursery, rules::ruff::rules::UnreachableCode),
         (Ruff, "015") => (RuleGroup::Unspecified, rules::ruff::rules::UnnecessaryIterableAllocationForFirstElement),
         (Ruff, "016") => (RuleGroup::Unspecified, rules::ruff::rules::InvalidIndexType),
+        (Ruff, "017") => (RuleGroup::Unspecified, rules::ruff::rules::TypeAliasAnnotation),
```

Comment:
I would still vote for `NonPEP695TypeAlias`. It's more consistent IMO, as it matches `NonPEP585Annotation` and `NonPEP604Annotation`.

Reply:
I'm tempted to say we should just put this in the `pyupgrade` category... We've added some other rules in there that don't originate in `pyupgrade` but instead fall into the same semantic category (code upgrades based on increasing Python version). What do you think?

Reply:
Sounds good to me.